sdk/eventhub/Azure.Messaging.EventHubs.Processor/tests/Processor/EventProcessorClientTests.cs

```diff
@@ -995,352 +991,51 @@ public async Task ProcessEventAsyncReceivesAnEmptyPartitionContextForNoData()
-
-            Assert.That(ownedByProcessor1.Count(), Is.EqualTo(MinimumpartitionCount));
-            Assert.That(ownedByProcessor1.Any(owned => claimablePartitionIds.Contains(owned.PartitionId)), Is.False);
-
-            // Start the processor to claim ownership from of a Partition even though ownedPartitionCount == MinimumOwnedPartitionsCount.
-
-            await processor.StartProcessingAsync(cancellationSource.Token);
-            await processor.WaitStabilization();
-
-            // Get owned partitions.
-
-            totalOwnedPartitions = await partitionManager.ListOwnershipAsync(processor.FullyQualifiedNamespace, processor.EventHubName, processor.ConsumerGroup, cancellationSource.Token);
-            ownedByProcessor1 = totalOwnedPartitions.Where(p => p.OwnerIdentifier == processor.Identifier);
-
-            // Verify that we took ownership of the additional partition.
-
-            Assert.That(ownedByProcessor1.Count(), Is.GreaterThan(MinimumpartitionCount));
-            Assert.That(ownedByProcessor1.Any(owned => claimablePartitionIds.Contains(owned.PartitionId)), Is.True);
-
-            await processor.StopProcessingAsync(cancellationSource.Token);
+            mockLoadbalancer.Verify(m => m.RelinquishOwnershipAsync(It.IsAny<CancellationToken>()));
         }
 
         /// <summary>
-        ///   Verifies that partitions ownership load balancing will direct an <see cref="EventProcessorClient" /> steal ownership of a partition
-        ///   from another <see cref="EventProcessorClient" /> the other processor owns greater than the calculated MaximumOwnedPartitionsCount.
+        ///   Verifies that partitions owned by an <see cref="EventProcessorClient" /> are immediately available to be claimed by another processor
+        ///   after <see cref="StopProcessingAsync" /> is called.
         /// </summary>
         ///
         [Test]
-        public async Task FindAndClaimOwnershipAsyncStealsPartitionsWhenThisProcessorOwnsMinPartitionsAndOtherProcessorOwnsGreatherThanMaxPartitions()
+        public async Task StartProcessingAsyncStartsLoadbalancer()
         {
-            const int MinimumpartitionCount = 4;
-            const int MaximumpartitionCount = 5;
-            const int NumberOfPartitions = 14;
-            Func<EventHubConnection> connectionFactory = () => new MockConnection();
-            MockConnection connection = connectionFactory() as MockConnection;
-            var partitionManager = new MockCheckPointStorage((s) => Console.WriteLine(s));
-            var processor = new MockEventProcessorClient(
-                partitionManager,
-                connectionFactory: connectionFactory,
-                numberOfPartitions: NumberOfPartitions,
-                clientOptions: default);
-
-            // Establish timed cancellation to ensure that the test doesn't hang.
-
             using var cancellationSource = new CancellationTokenSource();
```

Comment:
since `CancellationToken` is a struct, it won't work with `Mock.Of<CancellationToken>()`

Reply:
Would it make more sense to use `CancellationToken.None` in these cases, then?   I'd probably err on the side of setting a limit of 30 seconds or so unless we're sure that the path being tested has no async logic that could potentially hang.   We're likely being overly cautious for some of our unit tests, but I'd rather have an unnecessary timeout than try to figure out which test in a CI or nightly run was hanging.

Reply:
@jsquire That's fair. This makes me think that it would be nice to have some sort of test base class that enforces a per test timeout with a descriptive assert to address this problem globally.

Reply:
Not a bad idea.  Generally speaking, I'd rather see us do this with a custom extension for the test runner than introduce additional complexity for the tests themselves.   The less of a burden that we place on folks with custom framework-type things, I think the more approachable and understandable the project is.